### PR TITLE
[MRG + 1] Fix to topomap contours.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/4197

### CODE REVIEW

**Comment by Eric89GXL** [comment]
There is also this issue:

> Finally, if contours is 1, 2, 3, etc. I always get one more contour than expected (2, 3, 4, etc.).

Should we fix it here, too?

**Comment by jaeilepp** [comment]
I'm not sure if there's a clean way of doing that. See discussion at https://github.com/mne-tools/mne-python/pull/3797. Either the colorbar is going to look weird or the contour levels will be weird. IMO the way it is now works quite well for most of the cases. And if that's not good enough for the users, they can use custom values by passing an array.

**Comment by Eric89GXL** [comment]
Is there a good spot for doc clarification, then?

**Comment by jaeilepp** [comment]
Now I notice the issue is actually referencing to ``plot_tfr_topomap`` function, which has no contours param(?!). I'll take a better look tomorrow.

File: mne/viz/topomap.py [hunk 1]
```diff
@@ -1114,14 +1122,18 @@ def plot_tfr_topomap(tfr, tmin=None, tmax=None, fmin=None, fmax=None,
     cmap = _setup_cmap(cmap, norm=norm)
 
     if axes is None:
-        fig = plt.figure()
+        fig = plt.figure(figsize=(size, size))
```
Comment (agramfort) [inline mne/viz/topomap.py]: bug fix?

Comment (jaeilepp) [inline mne/viz/topomap.py]: Yeah.

**Comment by agramfort** [comment]
diff LGTM

can someone just test and report issues if any?

**Comment by jona-sassenhagen** [comment]
For me:

`evoked.plot_topomap(contours=False)` and `evoked.plot_topomap(contours=0)` both work.

`evoked.plot_topomap(contours=None)` gives
`TypeError: unorderable types: NoneType() > int()`

An empty list does not work:
`ValueError: zero-size array to reduction operation minimum which has no identity`

And it somehow isn't propagated to mne.plot_joint (`evoked.plot_joint(topomap_args={"contours":False})` gives 1 contour line)?

**Comment by jaeilepp** [comment]
IMO it's a bit silly to allow None for the contours. We already have options False and 0. I vote for removal of None option.

**Comment by jaeilepp** [comment]
And remove False too.

**Comment by jona-sassenhagen** [comment]
Fine with me.

**Comment by Eric89GXL** [comment]
@jona-sassenhagen are you happy now?

**Comment by jona-sassenhagen** [comment]
Can't check right now, on the move :(
The code looks good though. 
		

		
	





On Thu, Apr 20, 2017 at 4:10 PM +0200, "Eric Larson" <notifications@github.com> wrote:












@jona-sassenhagen are you happy now?

**Comment by Eric89GXL** [comment]
Works well over here and CI failure is just OSX being silly. Thanks @jaeilepp

**Comment by agramfort** [comment]
thx @jaeilepp and thx @jona-sassenhagen for testing.

**Comment by jona-sassenhagen** [comment]
Yes thanks @jaeilepp :D

**Pull request merged**